src/rhash/hash.c

```diff
@@ -3427,6 +3428,7 @@ int rc_hash_generate_from_file(char hash[33], uint32_t console_id, const char* p
     case RC_CONSOLE_VIRTUAL_BOY:
     case RC_CONSOLE_WASM4:
     case RC_CONSOLE_WONDERSWAN:
+    case RC_CONSOLE_ZX_SPECTRUM:
```

Comment:
Please add unit tests for at least `tap` and `tzx`. You can copy the Apple II or Commodore tests. Since Spectrum is using full file hash, it wouldn't be difficult to enumerate all supported extensions.